srcdeps-maven-local-repository/src/main/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManager.java

```diff
@@ -0,0 +1,260 @@
+import org.eclipse.aether.repository.LocalArtifactResult;
+import org.eclipse.aether.repository.LocalMetadataRegistration;
+import org.eclipse.aether.repository.LocalMetadataRequest;
+import org.eclipse.aether.repository.LocalMetadataResult;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.LocalRepositoryManager;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.l2x6.srcdeps.config.yaml.YamlConfigurationIo;
+import org.l2x6.srcdeps.core.BuildException;
+import org.l2x6.srcdeps.core.BuildRequest;
+import org.l2x6.srcdeps.core.BuildService;
+import org.l2x6.srcdeps.core.SrcVersion;
+import org.l2x6.srcdeps.core.config.BuilderIo;
+import org.l2x6.srcdeps.core.config.Configuration;
+import org.l2x6.srcdeps.core.config.ConfigurationException;
+import org.l2x6.srcdeps.core.config.ScmRepository;
+import org.l2x6.srcdeps.core.shell.IoRedirects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link LocalRepositoryManager} able to build the requested artifacts from their sources.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class SrcdepsLocalRepositoryManager implements LocalRepositoryManager {
+
+    /**
+     * Encapsulates the loading of {@link Configuration} that has to be done lazily - i.e. at some point when
+     * {@link MavenSession} is available already.
+     */
+    private static class LazyConfiguration {
+        private final Provider<MavenSession> sessionProvider;
+
+        public LazyConfiguration(Provider<MavenSession> sessionProvider) {
+            super();
+            this.sessionProvider = sessionProvider;
+        }
+
+        private volatile Configuration configuration;
+        private final Object configurationLock = new Object();
+
+        /**
+         * Loads the {@link Configuration} lazily. If multiple threads call this method while {@link #configuration} is
+         * uninitialized, only the first one loads from the file system while others are waiting.
+         *
+         * @return a {@link Configuration}
+         */
+        public Configuration getConfiguration() {
+            synchronized (configurationLock) {
+                if (configuration == null) {
+                    File srcdepsYaml = locateSrcdepsYaml();
+                    log.debug("SrcdepsLocalRepositoryManager using {}", srcdepsYaml.getAbsolutePath());
+                    try (Reader r = new InputStreamReader(new FileInputStream(srcdepsYaml), "utf-8")) {
+                        configuration = new YamlConfigurationIo().read(r);
+                    } catch (IOException | ConfigurationException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+            return configuration;
+        }
+
+        /**
+         * Finds the {@link ScmRepository} associated with the given {@code artifact}. The association is given by the
```

Comment:
Finds the *first* ... The code returns on the first match, so, if I have two entries for groupId `org.hawkular.accounts`, the first one is returned.